services/wallet/market/market.go

```diff
@@ -23,12 +23,21 @@ type DataPoint struct {
 	UpdatedAt int64
 }
 
+type MarketValuesSnapshot struct {
+	MarketValues thirdparty.TokenMarketValues
+	UpdatedAt    int64
+}
+
 type DataPerTokenAndCurrency = map[string]map[string]DataPoint
+type MarketValuesPerToken = map[string]MarketValuesSnapshot
+type MarketValuesPerCurrencyAndToken = map[string]MarketValuesPerToken
 
 type Manager struct {
 	feed            *event.Feed
 	priceCache      DataPerTokenAndCurrency
 	priceCacheLock  sync.RWMutex
+	marketCache     MarketValuesPerCurrencyAndToken
+	marketCacheLock sync.RWMutex
```

Comment:
Here we're adding the fields for `marketCache` and `marketCacheLock`. It seems a lock is needed for coordinating reads and writes from the cache.